### PR TITLE
Feat/live user

### LINKED ISSUE #25: Live tracking of users
- [x] Stream of current location updates 
- [x] Updates are pushed to firebase

DUE: End of 9/10

### PR DESCRIPTION
**Pull request by TsubasaK111**
- added .gitignore
- make componentDidMount async call
- subscribe to location by creating async function _getLocationAsync that emits callback and never `return`s.
- make user creation call a separate call

resolves #25 
review me!

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -68,69 +78,64 @@ export default class App extends React.Component {
...
       });
     } else {
-      this._getLocationAsync();
+      await this._setupUser();
+      await this._getLocationAsync();
     }
   }
 
+  _setupUser() {
+    if (this.state.user_id) {
```
Comment (DennoDin) [inline App.js]: Does this actually do anything? `this.state.user_id` starts as `null`, right, so this should not be executed?

File: App.js [hunk 2]
```diff
@@ -162,13 +167,40 @@ export default class App extends React.Component {
...
+    }
+
+    await Location.watchPositionAsync(
+      {
+        enableHighAccuracy: true,
+        distanceInterval: 1,
+        timeInterval: 200
+      },
+      location => {
+        console.log(location);
```
Comment (DennoDin) [inline App.js]: `console.log`

File: App.js (hunk 1 above)
Comment (DennoDin) [reply App.js]: It's not doing anything and the ID is being passed in as `null`

Comment (TsubasaK111) [reply App.js]: this makes sure that you don't make multiple create user calls to firebase.
there is a fix for id being passed in as null in #37 .

File: App.js (hunk 2 above)
Comment (TsubasaK111) [reply App.js]: yup!

**Pull request merged**